src/services/axios.ts

```
@@ -0,0 +1,149 @@
+
+    // Check if it's a 401 error and not a retry request already
+    // Also check if it's not the refresh token request itself that failed
+    if (error.response?.status === 401 && !originalRequest._retry && originalRequest.url !== REFRESH_TOKEN_URL) {
+
+      // If a refresh is already in progress, queue the original request
+      if (isRefreshing) {
+        return new Promise(function(resolve, reject) {
+          failedQueue.push({ resolve, reject });
+        }).then(token => {
+          // When refresh is done, retry the original request with the new token
+          originalRequest.headers['Authorization'] = 'Bearer ' + token;
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)